sqlglot/optimizer/simplify.py

```diff
@@ -0,0 +1,100 @@
+import sqlglot.expressions as exp
+
+
+TRUE = exp.Boolean(this=True)
+FALSE = exp.Boolean(this=False)
+NULL = exp.Null()
+
+
+def simplify(expression):
+    expression = expression.copy()
+
+    # wrap because you cannot replace a node without a parent
+    expression = exp.Paren(this=expression)
+
+    while True:
+        start = hash(expression)
+        expression = simplify_equality(expression)
+        expression = simplify_not(expression)
+        expression = simplify_conjunctions(expression)
+        expression = simplify_parens(expression)
+
+        if start == hash(expression):
+            break
+
+    return expression.this
+
+
+def simplify_equality(expression):
+    for node in reverse_traverse(expression, exp.EQ):
+        left = node.left
+        right = node.right
+
+        if NULL in (left, right):
+            node.replace(NULL)
+        elif left == right:
+            node.replace(TRUE)
+        elif (
+            isinstance(left, exp.Literal)
+            and isinstance(right, exp.Literal)
+            and left.is_string
```

Comment:
so this does not replace `1 = 2` with false? why is that?

Reply:
because number comparison is harder, i don't want to do it yet, we can add it later, for example

1.0 = 1